libraries/botframework-connector/src/auth/openIdMetadata.ts

```diff
@@ -51,11 +55,11 @@ export class OpenIdMetadata {
                 this.lastUpdated = new Date().getTime();
                 this.keys = (await getKeyResponse.json()).keys as IKey[];
             } else {
-                throw new Error(`Failed to load Keys: ${ getKeyResponse.status }`);
+                throw new AuthenticationError(`Failed to load Keys: ${ getKeyResponse.status }`, StatusCodes.BAD_REQUEST);
```

Comment:
If we are here is because we failed to load information relevant to the validation but it is not related to the request, and more to an error while trying to reach the intercom open id metadata endpoint. 500 might be more appropriate since the error is not due to a bad request.